### ISSUE TITLE
Add Orbital State Estimation Function

### ISSUE THREAD

**Issue by kylekrol** [pathfinder-for-autonomous-navigation/psim#78]
Depends on #65.

Add a CXX orbital state estimation function for GNC version 1.0.

In the far field, this function will provide position and velocity measurements for the leader and follower spacecrafts in ECI and ECEF. We will use a fixed step size integrator to propagate both orbits during each cycle of the flight computer.

If a timestamped position, velocity of the other satellite is received, it will be propagated forward to the present time with a variable step size integrator and replace the old position, velocity measurement of the old spacecraft.

In the near field, we will also have relative position, velocity in ECEF. Here we will add a simple state filter based on (most likely) Clohessy-Wiltshire dynamics to get a better relative velocity measurement. (@stewartaslan will open a separate issue for this filter - <link here>).

**Comment by nhz2** [comment]
TODO, figure out what to do with float vs fixed RTK data

**Comment by nhz2** [comment]
The estimator should also estimate the energy of both orbits.

**Comment by nhz2** [comment]
The estimator should respond to / predict maneuvers.

**Comment by nhz2** [comment]
Some issues with the current MATLAB orbit estimator implementation.
1. The orbit estimator can call the gravity function an unlimited number of times, this isn't possible on a teensy.
2. The estimator is numerically unstable.

To solve these I'm working on a redesigned orbit estimator, where the number of calls to the gravity function is controlled. Also in addition to the giant EKF the new filter will be multiple smaller filters running in parallel, each taking in different combinations of sensor readings/ ground data. That way if one EKF fails because of a bad sensor, the other filters can be used.
